cirq/api/google/v2/metrics.proto

```diff
@@ -0,0 +1,51 @@
+syntax = "proto3";
+
+package cirq.api.google.v1;
+
+option java_package = "com.google.cirq.api.google.v2";
+option java_outer_classname = "CalibrationProto";
+option java_multiple_files = true;
+
+// A snapshot of the performance metrics for a quantum processor at a
+// particular time.
+message MetricSnapshot {
+
+  // The time the metrics were collected, in unix time (milliseconds since
+  // Epoch minus leap seconds). Metric collection take time, so this is the
+  // time at which all of the metrics have been collected.
+  uint64 collection_time_ms = 1;
+
+  // All of the metrics collected during this snapshot.
+  repeated Metric metrics = 2;
+}
+
+// The actual metrics.
+message Metric {
+  // The name of the metric. This is the type of metric, i.e. 't1' or
+  // 'randomized_benchmarking'.
+  string name = 1;
+
+  // The targets of the metrics. Targets include things like the qubits (i.e.
+  // strings like `q0_1`), but may also be empty when the metric applies
+  // globally.
+  repeated string targets = 2;
+
+  // The values of this Metrics. Repeated to support multivalued metrics like
+  // tomography.
+  repeated Value values = 3;
+}
+
+// A generic metric value. Metrics of the same type should all support only
+// one of the types.
+message Value {
+  // The raw value.
+  oneof val {
```

Comment:
Maybe we could have the repetition in here, instead of in `Metric`? So you can have a `Union[..., List[Float]]` instead of a `List[Union[..., Float]]`.

Reply:
Repeated oneofs add a lot of indirection.  Plus it's not clear they need to be of the same type, actually.

Reply:
I've deleted the comment about same type.